### PR TITLE
replace jsoncpp with nlohmann::json

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will replace the jsoncpp library with nlohmann::json

### CODE REVIEW

**Comment by phlptp** [comment]
Resolve Issue #2652

**Comment by phlptp** [comment]
Woah all checks passed

File: src/helics/core/AsyncTimeCoordinator.hpp [hunk 1]
```diff
@@ -11,7 +11,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "CoreFederateInfo.hpp"
 #include "TimeDependencies.hpp"
 
-#include "json/forwards.h"
+#include "json/json_fwd.hpp"
```
Comment (nightlark) [inline src/helics/core/AsyncTimeCoordinator.hpp:14]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/BaseTimeCoordinator.cpp [hunk 2]
```diff
@@ -10,7 +10,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "flagOperations.hpp"
 #include "helics_definitions.hpp"
 
-#include "json/json.h"
+#include "json/json.hpp"
```
Comment (nightlark) [inline src/helics/core/BaseTimeCoordinator.cpp:13]:
```suggestion
#include <nlohmann/json.hpp>
```

File: src/helics/core/BaseTimeCoordinator.hpp [hunk 3]
```diff
@@ -10,7 +10,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "CoreFederateInfo.hpp"
 #include "TimeDependencies.hpp"
 
-#include "json/forwards.h"
+#include "json/json_fwd.hpp"
```
Comment (nightlark) [inline src/helics/core/BaseTimeCoordinator.hpp:13]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/BrokerBase.hpp [hunk 4]
```diff
@@ -16,6 +16,7 @@ and some common methods used cores and brokers
 #include "gmlc/containers/BlockingPriorityQueue.hpp"
 
 #include <atomic>
+#include <json/json_fwd.hpp>
```
Comment (nightlark) [inline src/helics/core/BrokerBase.hpp:19]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/ForwardingTimeCoordinator.cpp [hunk 5]
```diff
@@ -13,7 +13,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include <algorithm>
 #include <fmt/format.h>
 #include <iostream>
-#include <json/json.h>
+#include <json/json.hpp>
```
Comment (nightlark) [inline src/helics/core/ForwardingTimeCoordinator.cpp:16]:
```suggestion
#include <nlohmann/json.hpp>
```

File: src/helics/core/ForwardingTimeCoordinator.hpp [hunk 6]
```diff
@@ -11,9 +11,9 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "CoreFederateInfo.hpp"
 #include "TimeDependencies.hpp"
 
-#include "json/forwards.h"
 #include <atomic>
 #include <functional>
+#include <json/json_fwd.hpp>
```
Comment (nightlark) [inline src/helics/core/ForwardingTimeCoordinator.hpp:16]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/GlobalTimeCoordinator.cpp [hunk 7]
```diff
@@ -13,7 +13,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include <algorithm>
 #include <fmt/format.h>
 #include <iostream>
-#include <json/json.h>
+#include <json/json.hpp>
```
Comment (nightlark) [inline src/helics/core/GlobalTimeCoordinator.cpp:16]:
```suggestion
#include <nlohmann/json.hpp>
```

File: src/helics/core/GlobalTimeCoordinator.hpp [hunk 8]
```diff
@@ -11,9 +11,9 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "CoreFederateInfo.hpp"
 #include "TimeDependencies.hpp"
 
-#include "json/forwards.h"
 #include <atomic>
 #include <functional>
+#include <json/json_fwd.hpp>
```
Comment (nightlark) [inline src/helics/core/GlobalTimeCoordinator.hpp:16]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/InterfaceInfo.hpp [hunk 9]
```diff
@@ -12,7 +12,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "PublicationInfo.hpp"
 #include "gmlc/containers/DualMappedPointerVector.hpp"
 
-#include "json/forwards.h"
+#include "json/json_fwd.hpp"
```
Comment (nightlark) [inline src/helics/core/InterfaceInfo.hpp:15]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/TimeCoordinator.cpp [hunk 10]
```diff
@@ -10,10 +10,10 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "flagOperations.hpp"
 #include "helics_definitions.hpp"
 
-#include "json/json.h"
 #include <algorithm>
 #include <fmt/format.h>
 #include <iostream>
+#include <json/json.hpp>
```
Comment (nightlark) [inline src/helics/core/TimeCoordinator.cpp:16]:
```suggestion
#include <nlohmann/json.hpp>
```

File: src/helics/core/TimeCoordinator.hpp [hunk 11]
```diff
@@ -10,9 +10,9 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "ActionMessage.hpp"
 #include "BaseTimeCoordinator.hpp"
 
-#include "json/forwards.h"
 #include <atomic>
 #include <functional>
+#include <json/json_fwd.hpp>
```
Comment (nightlark) [inline src/helics/core/TimeCoordinator.hpp:15]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/TimeDependencies.cpp [hunk 12]
```diff
@@ -10,7 +10,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "ActionMessage.hpp"
 #include "flagOperations.hpp"
 
-#include "json/json.h"
+#include "json/json.hpp"
```
Comment (nightlark) [inline src/helics/core/TimeDependencies.cpp:13]:
```suggestion
#include <nlohmann/json.hpp>
```

File: src/helics/core/TimeDependencies.hpp [hunk 13]
```diff
@@ -8,7 +8,7 @@ SPDX-License-Identifier: BSD-3-Clause
 
 #include "basic_CoreTypes.hpp"
 
-#include "json/forwards.h"
+#include "json/json_fwd.hpp"
```
Comment (nightlark) [inline src/helics/core/TimeDependencies.hpp:11]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/helicsCLI11JsonConfig.hpp [hunk 14]
```diff
@@ -9,7 +9,7 @@ SPDX-License-Identifier: BSD-3-Clause
 
 #include "helicsCLI11.hpp"
 
-#include "json/forwards.h"
+#include <json/json_fwd.hpp>
```
Comment (nightlark) [inline src/helics/core/helicsCLI11JsonConfig.hpp:12]:
```suggestion
#include <nlohmann/json_fwd.hpp>
```

File: src/helics/core/queryHelpers.cpp [hunk 15]
```diff
@@ -12,61 +12,64 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "HandleManager.hpp"
 #include "InterfaceInfo.hpp"
 
+#include <json/json.hpp>
```
Comment (nightlark) [inline src/helics/core/queryHelpers.cpp:15]:
```suggestion
#include <nlohmann/json.hpp>
```

File: src/helics/shared_api_library/helicsData.h [hunk 16]
```diff
@@ -69,7 +69,7 @@ HELICS_EXPORT int32_t helicsDataBufferFillFromString(HelicsDataBuffer data, cons
 
 /** convert a raw string (may contain nulls) to serialized bytesin a buffer
 * @param data the HelicsDataBuffer to fill
-* @param value the value to serialize into the dataBuffer
+* @param str the value to serialize into the dataBuffer
 @return the buffer size after filling
```
Comment (nightlark) [inline src/helics/shared_api_library/helicsData.h:73]:
```suggestion
@param stringSize the size of the string value to serialize into the dataBuffer
@return the buffer size after filling
```

File: tests/helics/common/JsonGenerationTests.cpp [hunk 17]
```diff
@@ -8,6 +8,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "helics/common/JsonGeneration.hpp"
 #include "helics/common/JsonProcessingFunctions.hpp"
 
+#include "json/json.hpp"
```
Comment (nightlark) [inline tests/helics/common/JsonGenerationTests.cpp:11]:
```suggestion
#include <nlohmann/json.hpp>
```

File: tests/helics/system_tests/QueryTests.cpp [hunk 18]
```diff
@@ -18,6 +18,7 @@ SPDX-License-Identifier: BSD-3-Clause
 
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
+#include <json/json.hpp>
```
Comment (nightlark) [inline tests/helics/system_tests/QueryTests.cpp:21]:
```suggestion
#include <nlohmann/json.hpp>
```

**Review by nightlark — COMMENTED** [review]
Though currently we only support using a vendor copy of the nlohmann JSON library, we should make the include path match what it would be if we were using a system or package manager installed copy of the header files, `#include <nlohmann/json.hpp`.

File: CONTRIBUTORS.md [hunk 19]
```diff
@@ -65,9 +65,9 @@ Asio is used for TCP and UDP communication, as well as resolving IP addresses. T
 
 The header only bindings for the ZeroMQ library are used to interact with the ZeroMQ library. The header files are modified to include some string operations and are included in the HELICS source. cppzmq is licensed under the [MIT](https://github.com/zeromq/cppzmq/blob/master/LICENSE) license.
 
-### [JsonCpp](https://github.com/open-source-parsers/jsoncpp)
+### [Json](https://github.com/nlohmann/json)
```
[the remark this answers is not in the document]
Comment (nightlark) [reply CONTRIBUTORS.md:68]: Alternatives used to refer to it in package managers include `nlohmann-json`, and `nlohmann/json`.

**Pull request merged**